velox/exec/Task.cpp

```diff
@@ -1600,7 +1600,9 @@ void Task::addSplitLocked(
   ++taskStats_.numQueuedSplits;
 
   if (split.hasConnectorSplit()) {
-    VELOX_CHECK_NULL(split.connectorSplit->dataSource);
```

Comment:
In which case this would be violated?  If a data source is prefetched, it should not go through this path.

Reply:
In unit tests, a split is usually created and reused, for example:

https://github.com/facebookincubator/velox/blob/75b2349f2c40ac121941ad61a47beb683ab5bcc7/velox/exec/tests/TableScanTest.cpp#L413

Without this change, tests failed during adding split at below.

https://github.com/facebookincubator/velox/blob/75b2349f2c40ac121941ad61a47beb683ab5bcc7/velox/exec/Task.cpp#L1603

I was wondering if we can keep its reusability if it won't impact the runtime code path.